pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -234,11 +258,43 @@ func (pl *InterPodAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod
 		return framework.Queue, err
 	}
 
+	// When queuing this Pod:
+	// - 1. A new node is added with the pod anti-affinity key, the pod may become schedulable.
+	// - 2. The original node does not have the pod anti-affinity key but the modified node does, or vice versa, the pod may become schedulable.
+	// - 3. Both the original and modified nodes have the pod anti-affinity key and they differ, the pod may become schedulable.
 	for _, term := range antiTerms {
-		if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
-			logger.V(5).Info("a node with matched pod anti-affinity topologyKey was added/updated and it may make pod schedulable",
+		if originalNode == nil {
+			if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
+				// Case 1: A new node is added with the anti-affinity key.
+				logger.V(5).Info("A node with a matching pod anti-affinity topologyKey was added and it may make the pod schedulable",
+					"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+				return framework.Queue, nil
+			}
+			continue
+		}
+
+		originalTopologyValue, originalHasKey := originalNode.Labels[term.TopologyKey]
+		modifiedTopologyValue, modifiedHasKey := modifiedNode.Labels[term.TopologyKey]
+
+		if !originalHasKey && modifiedHasKey {
```

Comment:
```suggestion
		if originalHasKey != modifiedHasKey {
```

Why not just merge L279 and L286 (and the log?)

Reply:
This will make the code simpler. Thanks for the suggestion.